runtime/src/main/java/io/quarkus/vault/runtime/client/Private.java

```diff
@@ -10,7 +10,7 @@
 @Retention(RetentionPolicy.RUNTIME)
 public @interface Private {
 
-    class Literal extends AnnotationLiteral<Private> implements Private {
+    class Literal extends AnnotationLiteral<Private> {
```

Comment:
I think we want to keep this. This is the pattern we use for ``AnnotationLiteral``s (even if not recommended by IDEs).

Reply:
Agreed. This is how they're implemented; not just in Quarkus.